### PR TITLE
Move licenses to a folder named licenses (..)

### PR DESCRIPTION
**Pull request by Naim2000**
Closes https://github.com/mkw-sp/mkw-sp/issues/903. Man i spent up until 1am on this thing

### CODE REVIEW

**Comment by Naim2000** [comment]
Oh and just before I go to bed. What is this thing
image

File: payload/game/system/SaveManager.cc [hunk 1]
```diff
@@ -77,9 +77,17 @@ void SaveManager::initSPSave() {
...
+        wchar_t path[64], path_old[64];
+        swprintf(path_old, std::size(path_old), L"/mkw-sp/settings%u.ini", m_spLicenseCount);
+        swprintf(path, std::size(path), L"/mkw-sp/licenses/slot%u.ini", m_spLicenseCount);
+
+        auto oldLicenseExists = SP::Storage::Open(path_old, "r");
+        if (oldLicenseExists) {
+            oldLicenseExists->~FileHandle();
+            SP::Storage::CreateDir(L"/mkw-sp/licenses", true);
+            SP::Storage::Rename(path_old, path);
+            SP::Storage::Remove(path_old, false);
```
Comment (Naim2000) [inline payload/game/system/SaveManager.cc:89]: Why did i keep the remove()

**Comment by Naim2000** [comment]
surely we won't have to worry about the situation that both /mkw-sp/settingsX.ini and /mkw-sp/licenses/slotX.ini both exist right

File: payload/game/system/SaveManager.cc [hunk 2]
```diff
@@ -77,9 +77,16 @@ void SaveManager::initSPSave() {
...
-        wchar_t path[64];
-        swprintf(path, std::size(path), L"/mkw-sp/settings%u.ini", m_spLicenseCount);
-
+        wchar_t path[64], path_old[64];
+        swprintf(path_old, std::size(path_old), L"/mkw-sp/settings%u.ini", m_spLicenseCount);
+        swprintf(path, std::size(path), L"/mkw-sp/licenses/slot%u.ini", m_spLicenseCount);
+
+        auto oldLicenseExists = SP::Storage::Open(path_old, "r");
+        if (oldLicenseExists) {
+            oldLicenseExists->~FileHandle();
```
Comment (GnomedDev) [inline payload/game/system/SaveManager.cc:86]: Please do not manually call the destructor.

File: payload/game/system/SaveManager.cc [hunk 3]
```diff
@@ -77,9 +77,16 @@ void SaveManager::initSPSave() {
     char iniBuffer[2048];
 
     for (m_spLicenseCount = 0; m_spLicenseCount < std::size(m_spLicenses);) {
-        wchar_t path[64];
-        swprintf(path, std::size(path), L"/mkw-sp/settings%u.ini", m_spLicenseCount);
-
+        wchar_t path[64], path_old[64];
```
Comment (GnomedDev) [inline payload/game/system/SaveManager.cc:80]: pathOld

**Review by GnomedDev — CHANGES_REQUESTED** [review]

File: payload/game/system/SaveManager.cc (hunk 2 above)
Comment (Naim2000) [reply payload/game/system/SaveManager.cc]: You know I thought it was out of place as well. Except I literally have no idea how to do it otherwise

Comment (riidefi) [reply payload/game/system/SaveManager.cc]: It'll be called automatically when the variable goes out of scope

Comment (Naim2000) [reply payload/game/system/SaveManager.cc]: Oh. Well that was anti climatic

Comment (riidefi) [reply payload/game/system/SaveManager.cc]: You might want to do `= static_cast<bool>(...)` to force the temporary to be dropped before the rename

**Comment by riidefi** [comment]
LGTM, thanks!

**Pull request merged**